### ISSUE TITLE
Layout/LineLength doesn't count quotes/parentheses when over limit

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop/rubocop#11175]
Hello. :wave: I'm seeing strange behavior with Layout/LineLength which *I think* appeared in the most recent couple of versions where this cop won't catch excess line length issues when methods end in a quote, closing parentheses, and possibly other special characters.

## Expected behavior

I would expect the following to be flagged as a 100 character violation since it's 101 characters:

``` ruby
module Demo
  def self.one = "Pellentque morbi-trist sentus et netus et malesuada fames ac turpis egestas. Vest"
end
```

The problem is that the ending quote is on column 101 which is over the 100 column limit (if set as such).

## Actual behavior

First, configure RuboCop as follows:

``` yaml
Layout/LineLength:
  Max: 100
```

Here's demonstration code that highlights the issue:

``` ruby
module Demo
  # 101 characters.
  def self.one = "Pellentque morbi-trist sentus et netus et malesuada fames ac turpis egestas. Vest"

  # 102 characters.
  def self.two = "Pellentque morbi-trist sentus et netus et malesuada fames ac turpis egestas. Vestx"
end
```

Notice that the end quote (`"`) is not being counted. Only when an extra character is being added does this trigger the `Layout/LineLength` rule. The same situation happens with closing parentheses too.

## Steps to reproduce the problem

Use the configuration and code sample provided in the *Actual behavior* section.

## RuboCop version

```
ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [arm64-darwin21.6.0]

1.38.0 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 3.1.2) [arm64-darwin21.6.0]
  - rubocop-performance 1.15.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.15.0
  - rubocop-thread_safety 0.4.4
```

**Comment by fatkodima** [comment]
First example is 100 chars (not 101), and second is 101 (not 102).
```ruby
'  def self.one = "Pellentque morbi-trist sentus et netus et malesuada fames ac turpis egestas. Vest"'.size # => 100
```

I got a rubocop warning for second example, as expected.

**Comment by bkuhlmann** [comment]
Thanks. I'm closing this due to user error. I didn't realize what was going on until you pointed out the difference when testing this yourself. The problem appears to be with my editor (Sublime Text). It appears that the recent version of Sublime Text word wraps on the 99th column instead of the 100th which wasn't obvious to detect. This is a bug with Sublime Text and not RuboCop. Apologies.
